selfdrive/ui/ui.cc

```diff
@@ -79,9 +79,11 @@ void update_line_data(const UIState *s, const cereal::ModelDataV2::XYZTData::Rea
   *pvd = left_points + right_points;
 }
 
-void update_model(UIState *s, const cereal::ModelDataV2::Reader &model) {
+void update_model(UIState *s, 
+                  const cereal::ModelDataV2::Reader &model,
+                  const cereal::UiPlan::Reader &plan) {
   UIScene &scene = s->scene;
-  auto model_position = model.getPosition();
+  auto model_position = plan.getPosition();
   float max_distance = std::clamp(model_position.getX()[TRAJECTORY_SIZE - 1],
                                   MIN_DRAW_DISTANCE, MAX_DRAW_DISTANCE);
```

Comment:
This segfaults on old routes since the list will be empty. Not sure if we care about replaying old routes though. We might want to check that `uiPlan` is updated in the calling function, or check the length of `position` here.

Reply:
willdo